llpc/lower/PassRegistry.def

```diff
@@ -0,0 +1,42 @@
+/*
+ ***********************************************************************************************************************
+ *
+ *  Copyright (c) 2021 Advanced Micro Devices, Inc. All Rights Reserved.
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in all
+ *  copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ *
+ **********************************************************************************************************************/
+/**
+ ***********************************************************************************************************************
+ * @file  PassRegistry.def
+ * @brief LLPC header file: used as the registry of LLPC lowering passes
+ ***********************************************************************************************************************
+ */
+
+LLPC_PASS("llpc-spirv-lower-access-chain", SpirvLowerAccessChain())
+LLPC_PASS("llpc-spirv-lower-const-immediate-store", SpirvLowerConstImmediateStore())
+LLPC_PASS("llpc-spirv-lower-inst-meta-remove", SpirvLowerInstMetaRemove())
+LLPC_PASS("llpc-spirv-lower-terminator", SpirvLowerTerminator())
+LLPC_PASS("llpc-spirv-lower-translator", SpirvLowerTranslator())
+LLPC_PASS("llpc-spirv-lower-global", SpirvLowerGlobal())
+LLPC_PASS("llpc-spirv-lower-math-const-folding", SpirvLowerMathConstFolding())
+LLPC_PASS("llpc-spirv-lower-math-float-op", SpirvLowerMathFloatOp())
+LLPC_PASS("llpc-spirv-lower-memory-op", SpirvLowerMemoryOp())
+
+#undef LLVM_PASS
```

Comment:
Is this meant to be `LLPC_PASS`? Maybe the undef makes more sense in `registerPases` where the define is

Reply:
Yes that's meant to be LLPC_PASS thank, I'll update it.
It's common practice in the LLVM codebase to put the undef in the .def files to avoid having to write it every time we include the def file. So I let it there to stay consistent with the LLVM codebase.

Reply:
I see, that makes sense.